integrations/elevenlabs.md

```diff
@@ -0,0 +1,185 @@
+        twitter: andychert
+pypi: https://pypi.org/project/elevenlabs-haystack/
+repo: https://github.com/andychert/elevenlabs-haystack
+type: Model Provider
+report_issue: https://github.com/andychert/elevenlabs-haystack/issues
+logo: /logos/elevenlabs.png
+version: Haystack 2.0
+toc: true
+---
+
+### **Table of Contents**
+- [Overview](#overview)
+- [Installation](#installation)
+- [Usage](#usage)
+- [License](#license)
+
+## Overview
+
+This repository contains an integration of ElevenLabs' Text-to-Speech API with Haystack pipelines. This package allows you to convert text to speech using ElevenLabs' API and optionally save the generated audio to AWS S3.
+
+## Installation
+
+```bash
+pip install elevenlabs_haystack
+```
+
+## Usage
+
+#### **ElevenLabs API Key**
+
+To access the ElevenLabs API, you need to create an account and obtain an API key.
+
+1. Go to the [ElevenLabs](https://elevenlabs.ai/) website and sign up for an account.
+2. Once logged in, navigate to the **Profile** section.
+3. In the **API** section, generate a new API key.
+4. Copy the API key.
+
+#### **AWS Credentials**
+
+To store generated audio files on AWS S3, you need AWS credentials (Access Key ID, Secret Access Key) and specify a region.
+
+1. If you don’t have an AWS account, sign up at [AWS](https://aws.amazon.com/).
+2. Create a new IAM user and assign the necessary permissions to allow the user to upload files to S3. The `AmazonS3FullAccess` policy is sufficient for this example.
+3. Once the IAM user is created, download or note the **AWS Access Key ID** and **Secret Access Key**.
+4. Identify the **AWS Region** where your S3 bucket resides (e.g., `us-east-1`). This information can be found in the AWS Management Console.
+5. Finally, create or identify the S3 bucket where the generated audio files will be saved.
+
+Create a `.env` file in the root directory with the following content (replace with your actual credentials):
+
+```bash
+ELEVENLABS_API_KEY=sk_your_elevenlabs_api_key_here
+AWS_ACCESS_KEY_ID=your_aws_access_key_id
+AWS_SECRET_ACCESS_KEY=your_aws_secret_access_key
+AWS_REGION_NAME=us-east-1
+AWS_S3_BUCKET_NAME=your_s3_bucket_name
+```
+
+These variables will be automatically loaded using `dotenv` and used to access ElevenLabs and AWS services securely.
+
+### Basic Text-to-Speech Example
+
+This example shows how to use the `ElevenLabsTextToSpeech` component to convert text to speech and save the generated audio file locally or in an AWS S3 bucket. It uses environment variables to access sensitive credentials.
+
+```python
+from haystack.utils import Secret
+from elevenlabs_haystack import ElevenLabsTextToSpeech
+
+# Initialize the ElevenLabsTextToSpeech component using environment variables for sensitive data
+tts = ElevenLabsTextToSpeech(
+    elevenlabs_api_key=Secret.from_env_var("ELEVENLABS_API_KEY"),
+    output_folder="audio_files",  # Save the generated audio locally
+    voice_id="K8lgMMdmFr7QoEooafEf",  # ElevenLabs voice ID
```

Comment:
Maybe it was because of my credentials, but I had to change the `voice_id` and add `model_id` as well
```suggestion
    voice_id="9BWtsMINqrJLrRacOk9x",  # ElevenLabs voice ID
    model_id="eleven_multilingual_v2",
```

Reply:
Thanks, you can just specify a default value there, such as `Alice` voice. I've changed the code.

Reply:
Still not working for me. Raises error
```

ERROR:elevenlabs_haystack.component:Error during text-to-speech conversion: Missing required parameter: model_id

```

Reply:
Thank you for your patience. The version has been updated, checked, everything should work.

The problem with voice was because the example used premium voice.
`voice_id="Xb7hH8MSUJpSbSDYk0k2"` - default Alice